controllers/UsersController.js

```diff
@@ -0,0 +1,123 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import { User } from '../models';
+
+/** * Class representing users */
+export default class UsersController {
+  /**
+  * Register a user and return a JWT token
+  * @param {*} req - Request object
+  * @param {*} res - Response object
+  * @param {*} next - Next function
+  * @returns {token} token - JWT token
+  */
+  static registerUser(req, res, next) {
+    const { email, username, password } = req.body.user;
+    bcrypt.hash(password, 10, (err, hash) => {
+      if (err) {
+        return next(err);
+      }
+      User.findOrCreate({
+        where: {
+          email,
```

Comment:
I think you should consider checking for when either email OR username already exists, before it hits the db, right now it checks for only when both username and email already exists.

Reply:
Alright, thanks for spotting that out, however, the attributes `username` and `email` in the database model are set to `unique` which tells the database not to duplicate these fields, so if the user tries to duplicate them, the error is caught on `line 57` and a descriptive error message is thrown.

Reply:
@veeqtor , your making the username unique is affecting my social authentication, because a Google user can have one username and various emails. As a result, it crashes when same google user logs in with an account and tries to login with another email. We need users to be able to login with various accounts, I suggest you do not make username unique, since the email is already unique. Hence with each email, a user is unique.

Reply:
@OKiMaureen Don't worry it won't affect your work. We can have different ways around it. we can have a separate table for social or just store a unique identifier from the profile data sent from the third party sites or API. We'll discuss it tomorrow.

Reply:
@OKiMaureen We will discuss this tomorrow.